File: core/datacap-service/src/main/java/io/edurt/datacap/service/body/TableFilter.java
package io.edurt.datacap.service.body;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import io.edurt.datacap.common.sql.configure.SqlColumn;
import io.edurt.datacap.common.sql.configure.SqlType;
import io.edurt.datacap.spi.model.Pagination;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;
import java.util.Map;
import java.util.Properties;

@Data
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@SuppressFBWarnings(value = {"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
public class TableFilter
{
    private Pagination pagination;
    private List<OrderFilter> orders;
    private SqlType type = SqlType.SELECT;
    // If it is in preview mode, only the executed SQL is returned, and the specific content is executed
    private boolean preview = false;
    // If it is UPDATE | INSERT uses the following configuration
    private List<SqlColumn> columns;
    private Properties original;
    private String value;
    private ColumnFilter filter;
    private List<Map<String, String>> newColumns;
}

File: core/datacap-web/src/model/TableFilter.ts
import {OrderFilter} from "@/model/OrderFilter";
import {Pagination} from "@/entity/Pagination";
import {Operator} from "@/enum/Operator";

export class TableFilter
{
  pagination?: Pagination;
  orders?: Array<OrderFilter>;
  type?: SqlType;
  columns?: Array<SqlColumn>;
  original?: Map<string, string>;
  preview?: boolean;
  value?: string;
  filter: Filter;
  newColumns?: Array<any>;
}

export enum SqlType
{
  SELECT = 'SELECT',
  UPDATE = 'UPDATE',
  DELETE = 'DELETE',
  ALTER = 'ALTER',
  SHOW = 'SHOW',
  INSERT = 'INSERT',
  TRUNCATE = 'TRUNCATE'
}

export class SqlColumn
{
  column?: string;
  value?: string;
  original?: Map<string, object>;
}

export class ColumnFilter
{
  index: number;
  column: string;
  operator: string;
  operations: Array<Operator>;
  value: any
}

export class Filter
{
  condition: string;
  filters: Array<ColumnFilter>;
}
